serial_views.py

```diff
@@ -95,60 +138,80 @@ def ppr_view(request):
+        region_name = str(request.POST.get('region_name'))
         generation_type = str(request.POST.get('generation_type'))
         if LOCAL_TESTING is False:
             # stores the current region boundary
             res_powerplant_tbl = oep_models.ego_dp_res_classes['ResPowerPlant']
 
-        # select the shape of the region
-        wkbs = [Serializer.regions_wkbs[region_id]]
-        # Query the DB with the given wkbelement as input
-        for wkb in wkbs:
-            if LOCAL_TESTING is False:
-                # define the table columns for query
-                tbl_cols = Bundle(
-                    'powerplant',
-                    res_powerplant_tbl.id,
-                    res_powerplant_tbl.version,
-                    res_powerplant_tbl.generation_type,
-                    res_powerplant_tbl.generation_subtype,
-                    res_powerplant_tbl.scenario
-                )
-                # create query
-                # ToDo: Is there a way to apply ST_Transform to Bundle
-                oep_query = Serializer.session.query(
-                    res_powerplant_tbl.rea_geom_new.ST_Transform(4326),
-                    tbl_cols
-                ) \
-                    .filter(
-                        and_(
-                            # tbl_cols.c.rea_geom_new.ST_Transform(4326).ST_Within(wkb),
-                            res_powerplant_tbl.rea_geom_new.ST_Transform(4326).ST_Within(wkb),
-                            tbl_cols.c.version == EGO_DP_VERSION,
-                            tbl_cols.c.scenario == EGO_DP_SCENARIO,
-                            tbl_cols.c.generation_type == generation_type
-                        )
-                    ).limit(1000)
-
-                for record in oep_query:
-                    # region_contains = loadswkb(str(record.powerplant.rea_geom_new), True)
-                    region_contains = loadswkb(str(record[0]), True)
-                    feature = Feature(
-                        id=record.powerplant.id,
-                        geometry=region_contains,
-                        property=dict(
-                            generation_type=generation_type,
-                            generation_subtype=record.powerplant.generation_subtype
-                        )
+        region_nut = Serializer.regions_nuts[region_name]
+        if LOCAL_TESTING is False:
+            # define the table columns for query
+            tbl_cols = Bundle(
+                'powerplant',
+                res_powerplant_tbl.id,
+                res_powerplant_tbl.nuts,  # added
+                res_powerplant_tbl.version,
+                res_powerplant_tbl.generation_type,
+                res_powerplant_tbl.generation_subtype,
+                res_powerplant_tbl.scenario
+            )
+            # create query
+            if OEP_ACCESS == 'OEP_DIALECT':
+                # TODO find a way to convert the column rea_geom_new to srid 4326
```

Comment:
@christian-rli - do you know in what srid are the wkb in the table under the column rea_geom_new?

Reply:
QGIS tells me: srid=3035

Reply:
Can you find a way to get the transformation between 3035 to 4326 with shapely? *we can't unfortunately make use of ST_Transform of SQLAlchemy as oedialect does not handle it yet :(